File: tests/cpp/src/TestComPort.cpp
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

#include <iostream>
#include <string>
#include <vector>
#include <sstream>
#include <cameo/api/cameo.h>

using namespace std;
using namespace cameo;

int main(int argc, char *argv[]) {

	int numberOfTimes = 1;

	if (argc > 2) {
		numberOfTimes = stoi(argv[1]);
	}

	This::init(argc, argv);

	Server& server = This::getServer();

	int port = This::getCom().requestPort();
	cout << "Received port " << port << endl;

	// Loop the number of times.
	for (int i = 0; i < numberOfTimes; ++i) {

		// Start the application.
		unique_ptr<App> app = server.start("comportcpp");

		app->waitFor();

		cout << "Finished the application " << *app << endl;
	}

	port = This::getCom().requestPort();
	cout << "Received port " << port << " that must be greater than first port" << endl;

	return 0;
}

File: tests/java/src/eu/ill/cameo/test/TestComPort.java
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

package eu.ill.cameo.test;

import eu.ill.cameo.api.base.App;
import eu.ill.cameo.api.base.Server;
import eu.ill.cameo.api.base.This;


public class TestComPort {

	public static void main(String[] args) {
		
		This.init(args);
		
		int numberOfTimes = 1;
		if (args.length > 1) {
			numberOfTimes = Integer.parseInt(args[0]);
		}

		Server server = This.getServer();
		
		try {
			int port = This.getCom().requestPort();
			System.out.println("Received port " + port);
						
			// Loop the number of times.
			for (int i = 0; i < numberOfTimes; ++i) {
			
				// Start the application.
				App application = server.start("comportjava");
	
				application.waitFor();
				System.out.println("Finished application " + application);
			}
			
			port = This.getCom().requestPort();
			System.out.println("Received port " + port + " that must be greater than first port");
		}
		finally {
			This.terminate();
		}
		
		System.out.println("Finished the application");
	}

}